packages/core/src/data-editor/data-editor.tsx

```diff
@@ -180,40 +180,170 @@ function shiftSelection(input: GridSelection, offset: number): GridSelection {
     };
 }
 
-export interface Keybinds {
-    readonly selectAll: boolean;
-    readonly selectRow: boolean;
-    readonly selectColumn: boolean;
-    readonly downFill: boolean;
-    readonly rightFill: boolean;
-    readonly pageUp: boolean;
-    readonly pageDown: boolean;
-    readonly clear: boolean;
-    readonly copy: boolean;
-    readonly paste: boolean;
-    readonly cut: boolean;
-    readonly search: boolean;
-    readonly first: boolean;
-    readonly last: boolean;
+type Keybind = boolean | string;
+
+interface ForcedKeybinds {
+    copy: boolean;
+    cut: boolean;
+    paste: boolean;
+}
+
+export interface ConfigurableKeybinds {
+    readonly downFill: Keybind;
+    readonly rightFill: Keybind;
+    readonly clear: Keybind;
+    readonly closeOverlay: Keybind;
+    readonly acceptOverlayDown: Keybind;
+    readonly acceptOverlayUp: Keybind;
+    readonly acceptOverlayLeft: Keybind;
+    readonly acceptOverlayRight: Keybind;
+    readonly search: Keybind;
+    readonly delete: Keybind;
+    readonly activateCell: Keybind;
+
+    // Navigation Keybinds
+    readonly goToFirstColumn: Keybind;
+    readonly goToLastColumn: Keybind;
+    readonly goToFirstCell: Keybind;
+    readonly goToLastCell: Keybind;
+    readonly goToFirstRow: Keybind;
+    readonly goToLastRow: Keybind;
+    readonly goToNextPage: Keybind;
+    readonly goToPreviousPage: Keybind;
+
+    readonly goUpCell: Keybind;
+    readonly goDownCell: Keybind;
+    readonly goLeftCell: Keybind;
+    readonly goRightCell: Keybind;
+
+    readonly goUpCellRetainSelection: Keybind;
+    readonly goDownCellRetainSelection: Keybind;
+    readonly goLeftCellRetainSelection: Keybind;
+    readonly goRightCellRetainSelection: Keybind;
+
+    // Selection Keybinds
+    readonly selectToFirstColumn: Keybind;
+    readonly selectToLastColumn: Keybind;
+    readonly selectToFirstCell: Keybind;
+    readonly selectToLastCell: Keybind;
+    readonly selectToFirstRow: Keybind;
+    readonly selectToLastRow: Keybind;
+
+    readonly selectGrowUp: Keybind;
+    readonly selectGrowDown: Keybind;
+    readonly selectGrowLeft: Keybind;
+    readonly selectGrowRight: Keybind;
+
+    readonly selectAll: Keybind;
+    readonly selectRow: Keybind;
+    readonly selectColumn: Keybind;
 }
 
+type Keybinds = ConfigurableKeybinds & ForcedKeybinds;
+
+export type RealizedKeybinds = Readonly<Record<keyof ConfigurableKeybinds, string>> & ForcedKeybinds;
+
 const keybindingDefaults: Keybinds = {
-    selectAll: true,
```

Comment:
We should work on supporting these legacy binding setters as much as possible as well

Reply:
https://github.com/ProseMirror/prosemirror-commands/blob/master/src/commands.ts#L744
https://github.com/ProseMirror/prosemirror-keymap/blob/master/src/keymap.ts

Might be helpful: ProseMirror splits their keyboard shortcuts into two concepts, keymaps and commands.

Reply:
Yeah it's worth considering but requires a more complex setup. I'd be curious what the use case would be.